### PR TITLE
(chore) Add Turborepo

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [ ] My work includes tests or is validated by existing tests.

## Summary

[Turborepo](https://turborepo.org/docs) is a high-performance build system for JavaScript and TypeScript codebases. Turborepo provides superfast incremental builds with zero runtime overhead. 

Our average CI workflow run takes about 22 minutes. The latest [turbo run](https://github.com/openmrs/openmrs-esm-patient-chart/actions/runs/2036759336) took 9 minutes and 57 seconds. This is because build artefacts from the previous run got cached by turbo. When the next run happened, turbo restored the cache and replayed the cache contents instead of running the builds anew. Note especially how the build step took just 4 seconds.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -20,10 +20,8 @@ jobs:
         with:
           node-version: "14.x"
       - run: npx lerna bootstrap
-      - run: npx lerna run lint
-      - run: npx lerna run typescript
+      - run: npx turbo run lint typescript build
```
Comment (brandones) [inline .github/workflows/node.js.yml:23]: Unrelated to this PR: It strikes me as weird that the workflows are using `npx`/`npm` while the actual project uses `yarn`. We should probably switch it over.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "ci:publish": "lerna publish from-package --yes",
     "ci:prepublish": "lerna publish from-package --no-git-reset --yes --dist-tag next",
     "release": "lerna version --no-git-tag-version",
-    "verify": "lerna run lint && lerna run test && lerna run typescript",
+    "verify": "turbo run lint && turbo run typescript",
```
[the remark this answers is not in the document]
Comment (denniskigen) [reply package.json:13]: `lerna run test` doesn't actually work. We don't run the test script inside of each individual package. Rather we run it using the `test` script in the top level package.json

[the remark this answers is not in the document]
Comment (brandones) [reply package.json:13]: Yeah but so should we do like `turbo run lint && turbo run typescript && yarn:test`? And then this should be what husky runs pre-push.

[the remark this answers is not in the document]
Comment (denniskigen) [reply package.json:13]: Yep. Agree

**Comment by denniskigen** [comment]
> > Our average CI workflow run takes about 22 minutes (Compare with this Turbo-based run).
> 
> How fast is the Turbo-based run?

The latest [run](https://github.com/openmrs/openmrs-esm-patient-chart/actions/runs/2036759336) took 9 mins and 57 seconds. Note how the build step took a whopping 4 seconds!

**Review by brandones — APPROVED** [review]
Very cool, thank you @denniskigen !

**Pull request merged**